### ISSUE TITLE
[dlt] Convert DLT to a diffusers pipeline

### ISSUE THREAD

**Issue by shunk031** [creative-graphic-design/design-generators#25]
Convert DLT (ICCV 2023) into a complete diffusers-style package so a trained checkpoint can be loaded with `from_pretrained` and sampled for unconditional, category-conditioned, and category+size-conditioned layout generation without depending on the original training tree.

## Approach

- The original code already defines `dlt/models/dlt.py::DLT(ModelMixin, ConfigMixin)` with `@register_to_config` fields for `categories_num`, `latent_dim`, `num_layers`, `num_heads`, `dropout_r`, `activation`, `cond_emb_size`, and `cat_emb_size`, so the conversion should preserve the existing state dict and add the missing `DiffusionPipeline` and scheduler/config surface around it.
- No pretrained DLT weights are released in the README; the original workflow trains first with `python main.py --config configs/remote/dlt_publaynet_config.yaml --workdir <WORKDIR>`, then stores checkpoints under `logs/<workdir>/checkpoints/checkpoint-<epoch>/` using `accelerator.save_state(save_path)` plus `model.save_pretrained(save_path)`.
- Datasets must be prepared before training: RICO from `https://interactionmining.org/rico`, PubLayNet from `https://github.com/ibm-aur-nlp/PubLayNet`, and Magazine from `https://xtqiao.com/projects/content_aware_layout/`; the repo expects local paths in `dlt/configs/remote/dlt_*_config.py`, not bundled data.
- The main inference entry point is `dlt/generate_samples.py::sample_from_model`, which creates Gaussian initial boxes and a masked category tensor, then iterates `JointDiffusionScheduler.step_jointly()` for `num_cont_timesteps`; supported `--cond_type` values are `all`, `whole_box`, and `loc`.
- The custom scheduler lives in `dlt/diffusion.py::JointDiffusionScheduler(DDPMScheduler)` and combines continuous DDPM box noise with discrete category transition matrices; its default sampler uses temperature `0.8` and categorical sampling, so the new pipeline should expose temperature and deterministic sampling controls instead of leaving them hardcoded.
- Configs use `ml_collections.ConfigDict`; published defaults include seed `42`, `num_cont_timesteps=100`, `num_discrete_steps=10`, `beta_schedule="squaredcos_cap_v2"`, PubLayNet/RICO `num_epochs=800`, Magazine `num_epochs=2000`, and model defaults `latent_dim=512`, `num_layers=4`, `num_heads=8`.
- Dependencies are moderate but specific: Python 3.9, PyTorch 2.0.0, CUDA 11.7, `accelerate`, `diffusers["torch"]`, `ml_collections`, `labml_nn`, `einops`, `wandb`, OpenCV, and scientific Python packages from `environment.yml`.
- The recommended API remains diffusers, matching issue #2, but the actual task is not to rewrite the denoiser: implement `DLTPipeline(DiffusionPipeline)`, move `JointDiffusionScheduler` to a save/loadable `SchedulerMixin` or compatible config class, and package dataset-specific tokenizer/category metadata with the pipeline.

## Steps

- [ ] Prepare RICO, PubLayNet, and/or Magazine locally and train the original implementation because pretrained DLT weights are not released.
- [ ] Generate fixed-seed reference outputs with `dlt/generate_samples.py` for `all`, `whole_box`, and `loc` on at least one dataset checkpoint.
- [ ] Package `DLT` as the denoiser component, preserving strict state-dict compatibility with the original `save_pretrained` checkpoint directories.
- [ ] Implement a save/loadable scheduler for `JointDiffusionScheduler` parameters, including continuous timestep count, discrete timestep count, category vocabulary size, beta schedule, and temperature.
- [ ] Implement `DLTPipeline.__call__` for unconditional/category/category+size generation and expose seed/generator, temperature, batch size, and `cond_type`.
- [ ] Add no-weight unit tests for config round-trip, scheduler transition matrices, tensor shapes, and deterministic generation with a tiny random model.
- [ ] Add optional vendor parity tests comparing the new pipeline against `generate_samples.py` on a trained checkpoint with fixed seed.
- [ ] Convert trained checkpoints with `save_pretrained` and publish dataset-specific repos such as `creative-graphic-design/dlt-publaynet`, `creative-graphic-design/dlt-rico`, and `creative-graphic-design/dlt-magazine`.

## References

- Paper: https://arxiv.org/abs/2303.03755
- Weights: Not released by the original README; train with the original implementation first, then use `logs/<workdir>/checkpoints/checkpoint-<epoch>/`.
- Training data: RICO `https://interactionmining.org/rico`; PubLayNet `https://github.com/ibm-aur-nlp/PubLayNet`; Magazine `https://xtqiao.com/projects/content_aware_layout/`.
- Original implementation: vendor/dlt (Apache-2.0)

> Training policy: any training we run ourselves uses PyTorch Lightning (the `training` extra), with all training/evaluation configuration handled via [LightningCLI](https://lightning.ai/docs/pytorch/stable/cli/lightning_cli.html) (YAML configs + CLI overrides). See #2.

**Comment by creative-graphic-design-dev** [comment]
## Plan amendment: train-ourselves CLI and parity staging

DLT is a train-first v0.5 target. The converted package should use the shared
class-path-driven LightningCLI entry point:

```bash
uv run --package dlt --extra training \
  python -m traingen.lightning.cli fit \
  --config models/dlt/configs/training/dlt_publaynet.yaml
```

Do not add `dlt.training.cli` in the first PR unless a model-specific preflight
hook is required and documented. YAML files should carry full
`model.class_path` and `data.class_path` values.

The DLT vendor is Accelerate/plain PyTorch, so S0-S2 parity should compare a
local vendor-step adapter against `DLTTrainingModule.training_step`, not force
the vendor loop through Lightning. Required traces include `box`, `box_cond`,
`cat`, `mask_box`, `mask_cat`, `noise`, `t`, noised boxes/categories,
predictions, masked L2, masked CE, and total loss. S0 also compares DLT state
dicts plus scheduler `cont2disc` and transition matrices.

First PR scope: package skeleton, `DLT`, save/loadable
`DLTJointDiffusionScheduler`, processor, pipeline, tiny CPU tests, conversion
for vendor `save_pretrained` checkpoint directories, Lightning training smoke
YAMLs, and gated S0-S2 parity adapters. Full training, Hub publishing, Magazine
publication, and S5 equivalence are follow-ups.

Dataset scope should name the vendor-compatible RICO target `rico13`, not
`rico25`. Magazine remains gated until polygon/train-only handling is amended.

**Comment by creative-graphic-design-dev** [comment]
## PubLayNet S5 reproduction result

S5 evaluation is complete for PubLayNet at PR commit `bf4cd95`.

Protocol: vendor epoch 799 checkpoint vs package epoch 799 checkpoint, PubLayNet validation split, `all` conditioning, evaluation seeds `42`, `43`, and `44`. The package final checkpoint was regenerated from the saved Lightning checkpoint and completed to epoch 799 before evaluation.

| Metric | Vendor mean +/- std | Package mean +/- std | Delta (package - vendor) |
| --- | ---: | ---: | ---: |
| loss | `1.8069 +/- 0.0228` | `1.8775 +/- 0.0215` | `+0.0706` |
| masked L2 | `0.2883 +/- 0.0037` | `0.2977 +/- 0.0035` | `+0.0094` |
| masked CE | `0.3654 +/- 0.0047` | `0.3889 +/- 0.0047` | `+0.0234` |
| FID | `2.3806 +/- 0.0546` | `31.2920 +/- 0.1734` | `+28.9114` |
| overlap | `0.0241 +/- 0.0003` | `0.2130 +/- 0.0019` | `+0.1889` |
| IoU | `0.0034 +/- 0.0001` | `0.0448 +/- 0.0005` | `+0.0414` |
| alignment | `0.0102 +/- 0.0000` | `0.0258 +/- 0.0002` | `+0.0156` |

Training-loss curve check for the regenerated final segment (epochs 733-799): mean absolute train-loss delta is `0.4549`; final train losses are vendor `0.6560` and package `0.7291`. There is no monotonic late-epoch loss divergence, but generated layouts are substantially worse under the same sampling and metrics protocol.

Conclusion: PubLayNet training loss is close enough to rule out a gross topology/training-step mismatch, but full generation quality is not reproduced. The likely next debugging target is inference/sampling semantics or checkpoint state beyond the already-passing S0-S2 topology and training-step parity checks.

`models/dlt/TRAINING.md` now records this under dataset-specific Reproduction Results, with RICO13 and Magazine marked as not yet run.
